crates/utils/src/utils/validation.rs

```diff
@@ -82,14 +86,83 @@ pub fn is_valid_body_field(body: &Option<String>) -> LemmyResult<()> {
 }
 
 pub fn is_valid_bio_field(bio: &str) -> LemmyResult<()> {
-  let check = bio.chars().count() <= BIO_MAX_LENGTH;
-  if !check {
-    Err(LemmyError::from_message("bio_length_overflow"))
+  max_length_check(bio, BIO_MAX_LENGTH, String::from("bio_length_overflow"))
+}
+
+/// Checks the site name length, the limit as defined in the DB.
+pub fn site_name_length_check(name: &str) -> LemmyResult<()> {
+  min_max_length_check(
+    name,
+    SITE_NAME_MIN_LENGTH,
+    SITE_NAME_MAX_LENGTH,
+    String::from("site_name_required"),
+    String::from("site_name_length_overflow"),
+  )
+}
+
+/// Checks the site description length, the limit as defined in the DB.
+pub fn site_description_length_check(description: &str) -> LemmyResult<()> {
+  max_length_check(
+    description,
+    SITE_DESCRIPTION_MAX_LENGTH,
+    String::from("site_description_length_overflow"),
+  )
+}
+
+fn max_length_check(item: &str, max_length: usize, msg: String) -> LemmyResult<()> {
+  if item.len() > max_length {
+    Err(LemmyError::from_message(&msg))
+  } else {
+    Ok(())
+  }
+}
+
+fn min_max_length_check(
+  item: &str,
+  min_length: usize,
+  max_length: usize,
+  min_msg: String,
+  max_msg: String,
+) -> LemmyResult<()> {
+  if item.len() > max_length {
+    Err(LemmyError::from_message(&max_msg))
+  } else if item.len() < min_length {
+    Err(LemmyError::from_message(&min_msg))
   } else {
     Ok(())
   }
 }
 
+/// Attempts to build a regex and check it for common errors before inserting into the DB.
+pub fn build_and_check_regex(regex_str_opt: &Option<&str>) -> LemmyResult<Option<Regex>> {
+  regex_str_opt.map_or_else(
+    || Ok(None::<Regex>),
+    |regex_str| {
+      if regex_str.is_empty() {
+        // If the proposed regex is empty, return as having no regex at all; this is the same
+        // behavior that happens downstream before the write to the database.
+        return Ok(None::<Regex>);
+      }
+
+      RegexBuilder::new(regex_str)
+        .case_insensitive(true)
+        .build()
+        .map_err(|e| LemmyError::from_error_message(e, "invalid_regex"))
+        .and_then(|regex| {
+          // NOTE: It is difficult to know, in the universe of user-crafted regex, which ones
+          // may match against any string text. To keep it simple, we'll match the regex
+          // against an innocuous string - a single number - which should help catch a regex
+          // that accidentally matches against all strings.
+          if regex.is_match("1") {
```

Comment:
This would be problematic if there is ever a community that rejects numbers; however, I feel like this would work for the majority of cases. Not sure if there's a better way to check that wont get complicated quickly.

Reply:
This seems fine, a slur filter which matches a single character is almost certainly wrong.